crates/arrow_tools/src/record_batch.rs

```diff
@@ -80,12 +82,68 @@ pub fn try_cast_to(record_batch: RecordBatch, schema: SchemaRef) -> Result<Recor
     RecordBatch::try_new(schema, cols).context(UnableToConvertRecordBatchSnafu)
 }
 
+/// Flattens a list of struct types with a single field into a list of primitive types.
+/// The struct field must be a primitive type.
+///
```

Comment:
nit: I'd make it more explicit that for `List[Struct]` with multiple keys, all except the first will be ignored.

Reply:
Added, also covered with test